lib/src/deprecation.dart

```diff
@@ -220,11 +220,23 @@ enum Deprecation {
       );
 
   /// Returns the set of all deprecations done in or before [version].
-  static Set<Deprecation> forVersion(Version version) {
-    var range = VersionRange(max: version, includeMax: true);
+  static Set<Deprecation> forVersion(Version version, [Version? current]) {
```

Comment:
A comparison to the current version is not necessary here. This should just exclude any deprecation where `obsoleteIn` is non-null

Reply:
![xzMs8O1KQh17i](https://github.com/user-attachments/assets/89640c4b-cdf4-432c-b9d9-5b64ca4e35b1)